SDK/CalendarStoreController.swift

```diff
@@ -0,0 +1,148 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import UIKit
+import SchedJoulesApiClient
+
+public final class CalendarStoreController: UITabBarController {
+    /// Colors used by the SDK.
+    public struct ColorPalette {
+        public static let red = UIColor(red: 241/255.0, green: 102/255.0, blue: 103/255.0, alpha: 1)
+    }
+    
+    /// The API key to be used by the view controllers.
+    let apiKey: String
+    
+    /// The page identifier to be passed to the home page.
+    let pageIdentifier: String?
+    
+    /// The title to be used on the home page's navigation bar.
+    let homePageTitle: String?
+    
+    /// The global tint color used through out the SDK.
+    let tintColor: UIColor
+    
+    /** **For iOS 11.0+**
+     Set to false if you don't want to use large navigation bar titles.
+    */
+    let largeTitle: Bool
+    
+    // - MARK: Initialization
+    
+    // This method is only called when initializing a `UIViewController` from a `Storyboard` or `XIB`. The `CalendarStoreController` must only be used programatically, but every subclass of `UIViewController` must implement `init?(coder aDecoder: NSCoder)`.
+    public required init?(coder aDecoder: NSCoder) {
+        fatalError("CalendarStoreController must only be initialized programatically.")
+    }
+    
+    /**
+     - parameter apiKey: The API Key (access token) for the **SchedJoules API**.
+     - parameter pageIdentifier: The page identifier for the the home page.
+     - parameter title: The title for the `navigtaion bar` in the home page.
+     - parameter largeTitle: Set to `false` if you don't want to use large navigation bar titles.
+     - parameter tintColor: The tint color used through out the SDK, default is SchedJoules red.
+     */
+    public init(apiKey: String, pageIdentifier: String?, title: String?, largeTitle: Bool = true, tintColor: UIColor = ColorPalette.red) {
+        // Initialization
+        self.apiKey = apiKey
+        self.pageIdentifier = pageIdentifier
+        self.largeTitle = largeTitle
+        self.tintColor = tintColor
+        homePageTitle = title
+        super.init(nibName: nil, bundle: nil)
+        
+        // Set tab bar tint color
+        tabBar.tintColor = tintColor
+
+        // Array to hold the view controllers
+        var tabViewControllers = [UIViewController]()
+    
+        // Create home page with a specific page identifier
+        if pageIdentifier != nil {
```

Comment:
Both branches look pretty much the same to me, except that you pass a different `Query` instance to the ctor of `PageViewController`. How about just using a conditional expression instead?

Reply:
Better yet, you don't seem to use `pageIdentifier` anywhere else in this class. Do you really need to store it inside the instance? If not, why don't you replace the `pageIdentifier` argument of this ctor with the concrete `Query` instance? That way the convenience ctors would just pass an instance of `HomePageQuery` or `SinglePageQuery` and there is no need for a conditional statment at all. The correct `Query` would already be selected be the ctor that you use to instanciate this class.